providers/implementations/signature/dsa_sig.c

```diff
@@ -698,3 +864,228 @@ const OSSL_DISPATCH ossl_dsa_signature_functions[] = {
+    OSSL_FIPS_IND_SETTABLE_CTX_PARAM(OSSL_SIGNATURE_PARAM_FIPS_DIGEST_CHECK)
+    OSSL_FIPS_IND_SETTABLE_CTX_PARAM(OSSL_SIGNATURE_PARAM_FIPS_SIGN_CHECK)
+    OSSL_PARAM_END
+};
+
+static const OSSL_PARAM *dsa_sigalg_settable_ctx_params(void *vpdsactx,
+                                                        ossl_unused void *provctx)
+{
+    PROV_DSA_CTX *pdsactx = (PROV_DSA_CTX *)vpdsactx;
+
+    if (pdsactx != NULL && pdsactx->operation == EVP_PKEY_OP_VERIFYMSG)
+        return settable_sigalg_ctx_params;
+    return NULL;
+}
+
+static int dsa_sigalg_set_ctx_params(void *vpdsactx, const OSSL_PARAM params[])
+{
+    PROV_DSA_CTX *pdsactx = (PROV_DSA_CTX *)vpdsactx;
+    const OSSL_PARAM *p;
+
+    if (pdsactx == NULL)
+        return 0;
+    if (params == NULL)
+        return 1;
+
+    if (!OSSL_FIPS_IND_SET_CTX_PARAM(pdsactx, OSSL_FIPS_IND_SETTABLE0, params,
+                                     OSSL_SIGNATURE_PARAM_FIPS_KEY_CHECK))
+        return  0;
+    if (!OSSL_FIPS_IND_SET_CTX_PARAM(pdsactx, OSSL_FIPS_IND_SETTABLE1, params,
+                                     OSSL_SIGNATURE_PARAM_FIPS_DIGEST_CHECK))
+        return  0;
+    if (!OSSL_FIPS_IND_SET_CTX_PARAM(pdsactx, OSSL_FIPS_IND_SETTABLE2, params,
+                                     OSSL_SIGNATURE_PARAM_FIPS_SIGN_CHECK))
+        return 0;
+
+    p = OSSL_PARAM_locate_const(params, OSSL_SIGNATURE_PARAM_NONCE_TYPE);
+    if (p != NULL
+        && !OSSL_PARAM_get_uint(p, &pdsactx->nonce_type))
+        return 0;
+
+    if (pdsactx->operation == EVP_PKEY_OP_VERIFYMSG) {
+        p = OSSL_PARAM_locate_const(params, OSSL_SIGNATURE_PARAM_SIGNATURE);
+        if (p != NULL) {
+            OPENSSL_free(pdsactx->sig);
+            pdsactx->sig = NULL;
+            pdsactx->siglen = 0;
+            if (!OSSL_PARAM_get_octet_string(p, (void **)&pdsactx->sig,
+                                             0, &pdsactx->siglen))
+                return 0;
+        }
+    }
+    return 1;
+}
+
+#define IMPL_DSA_SIGALG(md, MD)                                         \
+    static OSSL_FUNC_signature_sign_init_fn dsa_##md##_sign_init;       \
+    static OSSL_FUNC_signature_sign_message_init_fn                     \
+        dsa_##md##_sign_message_init;                                   \
+    static OSSL_FUNC_signature_verify_init_fn dsa_##md##_verify_init;   \
+    static OSSL_FUNC_signature_verify_message_init_fn                   \
+        dsa_##md##_verify_message_init;                                 \
+                                                                        \
+    static int                                                          \
+    dsa_##md##_sign_init(void *vpdsactx, void *vdsa,                    \
+                         const OSSL_PARAM params[])                     \
+    {                                                                   \
+        static const char desc[] = "DSA Sigalg Sign Init";              \
```

Comment:
I would include the MD in the desc[] for all the Inits rather than SigAlg

Reply:
Sounds like a good idea!  Pitty you didn't notice this in the PR that refactored the RSA bits, sounds like I need to make a fixup PR for it.

Reply:
Done